src/util/audit.c

```diff
@@ -121,7 +121,11 @@ int util_audit_log(int type, const char *message, uid_t uid) {
         }
 
         if (audit_fd >= 0 && type != UTIL_AUDIT_TYPE_NOAUDIT) {
-                r = audit_log_user_avc_message(audit_fd, audit_type, message, NULL, NULL, NULL, uid);
+                if (type == UTIL_AUDIT_TYPE_AVC) {
+                        r = audit_log_user_avc_message(audit_fd, audit_type, message, NULL, NULL, NULL, uid);
+                } else {
+                        r = audit_log_user_message(audit_fd, audit_type, message, NULL, NULL, NULL, 1);
+                }
```

Comment:
Why are there two functions exposed by libaudit, anyway? Why doesn't `audit_log_user_avc_message()` just check whether the passed type is other than `AUDIT_USER_AVC` (or related AVC types), and then forward to `audit_log_user_message()`?

Reply:
Perfectly valid question, I believe it's for historical reasons. The implementation of both functions is very similar but they provide a slightly different output. Documentation says:

- **audit_log_user_avc_message** This function should be used by all apps that are SELinux object managers.
- **audit_log_user_message** This function should be used by all console apps that do not manipulate accounts or groups.

Reply:
Btw., can you elaborate on the last argument to `audit_log_user_message`? The man-page is not very helpful. I guess this is about whether this is an audit-message about a successful / expected operation, or whether this is about a failed / unexpected operation?

I think passing `1` unconditionally might not be right here, though.